### PR TITLE
(Windows) Set console to UTF-8 on init

### PR DESCRIPTION
**Pull request by anzz1**
From my testing this fixes the problems with UTF-8 inputs and outputs.

I dislike having to add this platform-specific code but it has to be done for it to work properly.
The only other option would be guiding users to go deep into settings to enable a "beta" UTF-8 feature...sigh.

At least is wrapped somewhat nicely as its' own function outside the main loop.

### CODE REVIEW

**Review by Green-Sky — COMMENTED** [review]
looks ok, but i am on linux only right now, so if someone on windows could confirm this works please....

**Comment by lofcz** [comment]
tested on win, LGTM. Considering the previous UTF-8 hack this replaces, I'd say it's better done this way.

File: main.cpp [hunk 1]
```diff
@@ -154,6 +156,32 @@ void sigint_handler(int signo) {
...
+    }
+    if (hConOut) {
+        // Enable ANSI colors on Windows 10+
+        if (con_use_color && !(dwMode & 0x4)) {
+            SetConsoleMode(hConOut, dwMode | 0x4); // ENABLE_VIRTUAL_TERMINAL_PROCESSING (0x4)
+        }
+        // Set console output codepage to UTF8
+        SetConsoleOutputCP(65001); // CP_UTF8
+    }
+    void* hConIn = GetStdHandle((unsigned long)-10); // STD_INPUT_HANDLE (-11)
```
Comment (j-f1) [inline main.cpp:177]: Should this be…

```suggestion
    void* hConIn = GetStdHandle((unsigned long)-10); // STD_INPUT_HANDLE (-10)
```

Comment (anzz1) [reply main.cpp]: Yes, indeed it should :smile:

**Review by ggerganov — APPROVED** [review]

**Comment by ggerganov** [comment]
Can merge after conflicts are resolved

**Comment by Estelle-gqy** [comment]
your comit contains main.cpp file, but the master branch I cloned does not contain this file, how can I fix the bug？

**Pull request merged**